homework/index.js

```diff
@@ -30,13 +30,163 @@
     return elem;
   }
 
+  function formatDateString(string) {
+    const date = new Date(string);
+    const options = {
+      weekday: 'long',
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+      hour: '2-digit',
+      minute: '2-digit',
+    };
+    return date.toLocaleDateString('en-GB', options);
+  }
+
+  function addRowToTable(parent, options = []) {
```

Comment:
This is a good attempt at making your code DRY (_Don't Repeat Yourself_ principle), however the implementation is overly complicated because you are adding support to cover the exceptional case of the first row, which requires a link instead of a text string. During the lecture I showed how you can write a simplified version, generic version and deal with exception elsewhere.